packages/redux/src/products/serverInitialState/__tests__/lists.test.ts

```diff
@@ -61,10 +74,42 @@ describe('products lists serverInitialState()', () => {
     expect(state.lists.hash).toBe(expectedHash);
   });
 
-  it('should initialize server state for a non product list', () => {
+  it('should build the correct hash when is a custom listing page', () => {
+    const slug = '/en-pt/customlistingpage';
+    const expectedHash = 'listing/customlistingpage';
+    // @ts-expect-error A lot of properties would need to be added to make the value comply with the type which are irrelevant for the test
+    const model = {
+      relatedCommerceData: { referencedListing: [mockProductsListModel] },
+      slug,
+    } as Model;
+    const state = listsServerInitialState({ model });
+
+    expect(Object.keys(state.entities!.productsLists!)).toEqual(
+      expect.arrayContaining([expectedHash]),
+    );
+    expect(state.lists.hash).toBe(expectedHash);
+  });
+
+  it('should initialize server state for a non product list with initial state', () => {
     const model = {} as Model;
     const state = listsServerInitialState({ model });
 
     expect(state).toMatchSnapshot();
   });
+
+  it('should initialize server and throw a error for a non product list and is a listing', () => {
```

Comment:
This description is a little bit confusing. I think you mean something like this, but correct me if it is not the case:

 ```suggestion
  it('should initialize server state for listing that has returned no products and it is not a custom listing page', () => {
```